app/controllers/staff/events_controller.rb

```diff
@@ -2,6 +2,7 @@ class Staff::EventsController < Staff::ApplicationController
   before_action :enable_staff_subnav
 
   def edit
+    authorize @event, :edit?
```

Comment:
This will keep anyone not authorized from bringing up the edit page. However, the update actions are what processes any changes, so it's prudent to put auth checks on that and any action that can make organizer-only changes to the event. Also, any edit/save buttons that submit data to those actions should probably be disabled or hidden too.